plugins/inputs/pgbouncer/pgbouncer.go

```diff
@@ -0,0 +1,168 @@
+  ##       ?sslmode=[disable|verify-ca|verify-full]
+  ## or a simple string:
+  ##   host=localhost user=pqotest password=... sslmode=... dbname=app_production
+  ##
+  ## All connection parameters are optional.
+  ##
+  ## Without the dbname parameter, the driver will default to a database
+  ## with the same name as the user. This dbname is just for instantiating a
+  ## connection with the server and doesn't restrict the databases we are trying
+  ## to grab metrics for.
+  ##
+  address = "host=localhost user=pgbouncer sslmode=disable"
+`
+
+func (p *PgBouncer) SampleConfig() string {
+	return sampleConfig
+}
+
+func (p *PgBouncer) Description() string {
+	return "Read metrics from one or many pgbouncer servers"
+}
+
+func (p *PgBouncer) IgnoredColumns() map[string]bool {
+	return ignoredColumns
+}
+
+func (p *PgBouncer) Gather(acc telegraf.Accumulator) error {
+	var (
+		err     error
+		query   string
+		columns []string
+	)
+
+	query = `SHOW STATS`
+
+	rows, err := p.DB.Query(query)
+	if err != nil {
+		return err
+	}
+
+	defer rows.Close()
+
+	// grab the column information from the result
+	if columns, err = rows.Columns(); err != nil {
+		return err
+	}
+
+	for rows.Next() {
+		err = p.accRow(rows, acc, columns)
+		if err != nil {
+			return err
+		}
+	}
+
+	query = `SHOW POOLS`
+
+	poolRows, err := p.DB.Query(query)
+	if err != nil {
+		return err
+	}
+
+	defer poolRows.Close()
+
+	// grab the column information from the result
+	if columns, err = poolRows.Columns(); err != nil {
+		return err
+	}
+
+	for poolRows.Next() {
+		err = p.accRow(poolRows, acc, columns)
+		if err != nil {
+			return err
+		}
+	}
+
+	return poolRows.Err()
+}
+
+type scanner interface {
+	Scan(dest ...interface{}) error
+}
+
+func (p *PgBouncer) accRow(row scanner, acc telegraf.Accumulator, columns []string) error {
+	var columnVars []interface{}
+	var dbname bytes.Buffer
+
+	// this is where we'll store the column name with its *interface{}
+	columnMap := make(map[string]*interface{})
+
+	for _, column := range columns {
+		columnMap[column] = new(interface{})
+	}
+
+	// populate the array of interface{} with the pointers in the right order
+	for i := 0; i < len(columnMap); i++ {
+		columnVars = append(columnVars, columnMap[columns[i]])
+	}
+
+	// deconstruct array of variables and send to Scan
+	err := row.Scan(columnVars...)
+
+	if err != nil {
+		return err
+	}
+	if columnMap["database"] != nil {
+		// extract the database name from the column map
+		dbname.WriteString((*columnMap["database"]).(string))
+	} else {
+		dbname.WriteString("postgres")
+	}
+
+	var tagAddress string
+	tagAddress, err = p.SanitizedAddress()
+	if err != nil {
+		return err
+	}
+
+	tags := map[string]string{"server": tagAddress, "db": dbname.String()}
+
+	fields := make(map[string]interface{})
+	for col, val := range columnMap {
+		_, ignore := ignoredColumns[col]
+		if !ignore {
+			fields[col] = *val
+		}
+	}
+	acc.AddFields("pgbouncer", fields, tags)
```

Comment:
I think it will be best to have two measurements: `pgbouncer` and `pgbouncer_pools` since the values are unrelated.